### PR TITLE
add xdp dependencies for publish command

### PR DESCRIPTION
**Pull request by ami-GS**
## Description

add xdp dependencies to fpm

### CODE REVIEW

File: scripts/make-packages.sh [hunk 1]
```diff
@@ -174,25 +174,63 @@ if [ "$OS" == "linux" ]; then
...
+        --conflicts ${CONFLICTS} \
+        --depends "libssl${TLSVERSION}" \
+        --depends "libnuma1" \
+        --depends "libxdp1" \
+        --depends "libbpf1" \
+        --depends "libnl-3-200" \
+        --depends "libnl-route-3-200" \
+        --depends "libelf1t64" \
+        --depends "libz3-4" \
+        --depends "libzstd1" \
```
Comment (nibanks) [inline scripts/make-packages.sh:199]: All these are for xdp?

File: scripts/make-packages.sh [hunk 2]
```diff
@@ -174,25 +174,63 @@ if [ "$OS" == "linux" ]; then
...
-    --license MIT \
-    --url https://github.com/microsoft/msquic \
-    --log error \
-    ${FILES}
+
+  DONE_DEBIAN=0
+  if [ -e /etc/os-release ]; then
+    source /etc/os-release
+    # Ubuntu 24.04 support xdp
+    if [ "$ID" == "ubuntu" ] && [ "$VERSION_ID" == "24.04" ]; then
```
Comment (nibanks) [inline scripts/make-packages.sh:182]: This is a problem... we only build on ubuntu 24.04, but publish for the other distros. We aren't building a version of 24.04 **without** XDP for those other distros. We need to, or they will break.

File: scripts/make-packages.sh (hunk 1 above)
Comment (ami-GS) [reply scripts/make-packages.sh:199]: Yes!
https://github.com/microsoft/msquic/blob/9ebfd0d1cd776be52bfd686ba341f94ef80e8a73/src/platform/CMakeLists.txt#L100

File: scripts/make-packages.sh (hunk 2 above)
Comment (ManickaP) [reply scripts/make-packages.sh:182]: Should this rather be a script argument, same way as openssl 1.1 vs 3 is?

Comment (csujedihy) [reply scripts/make-packages.sh:182]: This should be a script argument. The environment where we publish packages isn't necessarily the environment where the packages are built. It should never be. We build packages in a different pipeline and reference them in this publishing pipeline.

Comment (csujedihy) [reply scripts/make-packages.sh:182]: Also, we should only publish ubuntu 24.04 packages to ubuntu 24.04 repo. The other distros that use openssl3 should use packages built on 22.04 and that use openssl should use packages built on 20.04, i.e. same as before.

File: scripts/make-packages.sh [hunk 3]
```diff
@@ -129,34 +129,45 @@ echo "ARCH=$ARCH PKGARCH=$PKGARCH ARTIFACTS=$ARTIFACTS"
 mkdir -p ${OUTPUT}
 
 if [ "$OS" == "linux" ]; then
-  # RedHat/CentOS
-  FILES="${ARTIFACTS}/libmsquic.${LIBEXT}.${VER_MAJOR}.${VER_MINOR}.${VER_PATCH}=/usr/${LIBDIR}/libmsquic.${LIBEXT}.${VER_MAJOR}.${VER_MINOR}.${VER_PATCH}"
-  FILES="${FILES} ${ARTIFACTS}/libmsquic.${LIBEXT}.${VER_MAJOR}=/usr/${LIBDIR}/libmsquic.${LIBEXT}.${VER_MAJOR}"
-  if [ -e "$ARTIFACTS/libmsquic.lttng.${LIBEXT}.${VER_MAJOR}.${VER_MINOR}.${VER_PATCH}" ]; then
-     FILES="${FILES} ${ARTIFACTS}/libmsquic.lttng.${LIBEXT}.${VER_MAJOR}.${VER_MINOR}.${VER_PATCH}=/usr/${LIBDIR}/libmsquic.lttng.${LIBEXT}.${VER_MAJOR}.${VER_MINOR}.${VER_PATCH}"
+  IsUbuntu2404=0
+  if [ -e /etc/os-release ]; then
```
Comment (csujedihy) [inline scripts/make-packages.sh:133]: Why are we still determining package version by looking at this config file on the publishing machine? It isn't the same machine the packages are built.

**Review by nibanks — APPROVED** [review]

**Pull request merged**